src/modules/sv_detection_methods/analyze_sa_tag_method.cpp

```diff
@@ -52,27 +52,28 @@ void retrieve_aligned_segments(std::string sa_string, std::vector<AlignedSegment
 
 void analyze_aligned_segments(const std::vector<AlignedSegment> & aligned_segments,
                               std::vector<Junction> & junctions,
+                              const seqan3::dna5_vector & query_sequence,
                               const std::string & read_name)
 {
     for(size_t i = 1; i<aligned_segments.size(); i++)
     {
         AlignedSegment current = aligned_segments[i-1];
         AlignedSegment next = aligned_segments[i];
         int32_t distance_on_read = next.get_query_start() - current.get_query_end();
-        // Neither gap nor overlap on read
-        if (distance_on_read >= -10 && distance_on_read <= 10)
+        // Check that there is neither an overlap nor a larger gap on the read
+        // TODO(eldarion): add command-line parameters for changing these currently hard-coded cutoffs
+        if (distance_on_read >= 0 && distance_on_read <= 10)
         {
             Breakend mate1{current.ref_name,
-                            current.orientation == strand::forward ? current.get_reference_end()
-                                                                   : current.get_reference_start(),
-                            current.orientation,
-                            sequence_type::reference};
+                           current.orientation == strand::forward ? current.get_reference_end()
+                                                                  : current.get_reference_start(),
+                           current.orientation};
             Breakend mate2{next.ref_name,
-                            next.orientation == strand::forward ? next.get_reference_start()
-                                                                : next.get_reference_end(),
-                            next.orientation,
-                            sequence_type::reference};
-            Junction new_junction{std::move(mate1), std::move(mate2), read_name};
+                           next.orientation == strand::forward ? next.get_reference_start()
+                                                               : next.get_reference_end(),
+                           next.orientation};
+            auto inserted_bases = query_sequence | seqan3::views::slice(current.get_query_end(), next.get_query_start());
+            Junction new_junction{std::move(mate1), std::move(mate2), std::move(inserted_bases), read_name};
```

Comment:
Do you need to do std::move here? Isn't there a std::move in the constructor as well?

Reply:
Thanks for this comment. It turned out I just was not aware of what `std::move` actually does. You are completely right and I remove the `std::move` around `mate1`, `mate2` and `inserted_bases`.
I also made the `inserted_sequence` parameter in the `Junction` constructor const reference because it is only used as an input to a pipeline of views.